classes/sota_sanity.bbclass

```diff
@@ -10,6 +10,32 @@ def sota_check_required_variables(status, d):
         if not d.getVar(var):
             status.addresult("%s should be set in your local.conf.\n" % var)
 
+def sota_check_variables_validity(status, d):
+    var = d.getVar("OSTREE_BRANCHNAME")
+    if var != "":
+        for ch in var:
+            if not (ch >= 'a' and ch <= 'z' or ch >= 'A' and ch <= 'Z' or ch >= '0' and ch <= '9' or ch = '_' or ch == '-'):
+                status.addresult("OSTREE_BRANCHNAME Should only contain characters from the character set [a-zA-Z0-9_-].\n")
+                break
+    var = d.getVar("{SOTA_HARDWARE_ID")
+    if var != "":
+        for ch in var:
+            if not (ch >= 'a' and ch <= 'z' or ch >= 'A' and ch <= 'Z' or ch >= '0' and ch <= '9' or ch = '_' or ch == '-'):
+                status.addresult("SOTA_HARDWARE_ID Should only contain characters from the character set [a-zA-Z0-9_-].\n")
+                break
+    var = d.getVar("SOTA_CLIENT_FEATURES")
+    if var != "hsm" and var != "secondary-network" and var != "":
```

Comment:
more consise: `if d.getVar("SOTA_CLIENT_FEATURES") not in ("hsm", "secondary-network", ""):`